From: Sven Roederer <devel-sven@geroedel.de>
Date: Thu, 22 Apr 2021 22:35:58 +0200
Subject: luci.mk: make SUBMENU package define customizable and optional

When including luci.mk in external repos it'S sometimes usefull to not use
the default LUCI-submenu hierarchy.
This change defines the LUCI_MENU variable which completely overrides the
default submenu of the LuCI config-section. When LUCI_MENU is not defined,
the default submenu derrived from LUCI_TYPE or "Application" fallback is used.
Defining LUCI_MENU in the package Makefile will just use this value. Setting
LuCI_MENU to and empty value will not define a submenu at all.
Together with LUCI_SECTION and LUCI_CATEGORY menu items can now created at any
place.

Signed-off-by: Sven Roederer <devel-sven@geroedel.de>

diff --git a/luci.mk b/luci.mk
index 319551789f221630c574c190babab05cfe8a040c..f7a7658baa34669f73d62e8c558aff15bac4dda4 100644
--- a/luci.mk
+++ b/luci.mk
@@ -123,10 +123,15 @@ PKG_GITBRANCH?=$(if $(DUMP),x,$(strip $(shell \
 
 include $(INCLUDE_DIR)/package.mk
 
+LUCI_SUBMENU_DEFAULT=$(if $(LUCI_MENU.$(LUCI_TYPE)),$(LUCI_MENU.$(LUCI_TYPE)),$(LUCI_MENU.app))
+LUCI_SUBMENU=$(if $(LUCI_MENU),$(LUCI_MENU),$(LUCI_SUBMENU_DEFAULT))
+
 define Package/$(PKG_NAME)
   SECTION:=$(LUCI_SECTION)
   CATEGORY:=$(LUCI_CATEGORY)
-  SUBMENU:=$(if $(LUCI_MENU.$(LUCI_TYPE)),$(LUCI_MENU.$(LUCI_TYPE)),$(LUCI_MENU.app))
+ifneq ($(LUCI_SUBMENU),none)
+  SUBMENU:=$(LUCI_SUBMENU)
+endif
   TITLE:=$(if $(LUCI_TITLE),$(LUCI_TITLE),LuCI $(LUCI_NAME) $(LUCI_TYPE))
   DEPENDS:=$(LUCI_DEPENDS)
   VERSION:=$(if $(PKG_VERSION),$(PKG_VERSION),$(PKG_SRC_VERSION))
